### DISCUSSION TITLE
定时触发支持设置不同参数

### DISCUSSION THREAD

**Discussion by zanyzhao** [Tencent/bk-ci#4134] [Ideas]
目前的定时参数设置如下：
![image](https://user-images.githubusercontent.com/1629409/117523536-83e60f80-afeb-11eb-8e04-bd16d9bfa13c.png)

看下如何更好的支持不同参数

**Comment by jerrybean** [comment]
希望支持一定时间的 random，在设置的定时任务范围内random执行。如设置每日上午8点 random 30min，则只需要 8:00 - 8:30 内随机时间执行即可。
方便将定时任务打散，避免同一流水线的很多实例集中执行，对于下游系统如拉取代码、编译资源、发布系统造成过大压力。

**Reply by zanyzhao** [reply]
random不一定能100%解决并发大的问题，是不是应该让下游系统本身考虑限流会更好一些？

**Reply by jerrybean** [reply]
是的，但是下游限流会导致流水线某些插件执行耗时过大。我们目前做法是在定时流水线第一个stage加个random sleep 来执行。
或许 #4073 可以解决我们的场景
